### PR TITLE
More unit tests

### PR DESCRIPTION
**Pull request by jciechowski**
More simple tests.
I'm thinking about testing protected methods from Endpoint class but have to think how to do that.

### CODE REVIEW

**Comment by shrayasr** [comment]
Thanks! 😄 I'm currently travelling and I'll check this once I get back home.

File: pinboard.net.UnitTests/YesNoConverterSpec.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        [InlineData(true, "yes")]
+        public void ShouldWriteCorrectValue(object value, string expected)
+        {
+            var jsonWriterMock = new Mock<JsonWriter>();
+            jsonWriterMock.Setup(m => m.WriteValue(It.IsAny<string>()));
+            var sut = new YesNoConverter();
+
+            sut.WriteJson(jsonWriterMock.Object, value, null);
+
+            jsonWriterMock.Verify(m => m.WriteValue(expected));
```
Comment (shrayasr) [inline pinboard.net.UnitTests/YesNoConverterSpec.cs:35]: This test here doesn't have an `Assert` call. So a failure of this test will be a failure of the `Verify` call. Am I right? 

Unsure if that is the right behaviour?

Comment (jciechowski) [reply pinboard.net.UnitTests/YesNoConverterSpec.cs:35]: Yes, you are right. Basically it's a Moq method to verify if the method was called with expected values. 

For example look at this [WriteJson](https://github.com/shrayasr/pinboard.net/blob/master/pinboard.net/Util/Converters/YesNoConverter.cs) method - it doesn't return any values nor change state so the only way to check if it is behaving as designed is to verify if it was called with expected values and that's exactly what the Verify method is doing.

Comment (shrayasr) [reply pinboard.net.UnitTests/YesNoConverterSpec.cs:35]: Got it 👍

**Comment by shrayasr** [comment]
Thank you 🙏

**Pull request merged**